src/protocol/farcaster/bundler.ts

```diff
@@ -0,0 +1,83 @@
+import { Interface } from 'ethers/lib/utils';
+import { Transaction } from '../../types';
+import { FarcasterContracts } from './constants';
+
+export const bundlerContextualizer = (
+  transaction: Transaction,
+): Transaction => {
+  const isBundler = detectBundler(transaction);
+  if (!isBundler) return transaction;
+
+  return generateBundlerContext(transaction);
+};
+
+export const detectBundler = (transaction: Transaction): boolean => {
+  if (transaction.to !== FarcasterContracts.Bundler.address) {
+    return false;
+  }
+
+  try {
+    const iface = new Interface(FarcasterContracts.Bundler.abi);
+    const decoded = iface.parseTransaction({
+      data: transaction.input,
+      value: transaction.value,
+    });
+
+    return ['register'].includes(decoded.name);
```

Comment:
This could be more readable if it were inverted
```suggestion
    return decoded.name.includes('register');
```

Reply:
Never mind, I see how this approach is better in other contextualizer files like this: `return ['changeRecoveryAddressFor', 'transfer'].includes(decoded.name);`

Reply:
Wouldn't just `decoded.name === 'register'` be better then? The reason it does a "does the array include this value"-check here is that all farcaster detect-functions currently use the same pattern and most check for more than one function (like `['functionA', 'functionB'].includes(decoded.name)`)

Reply:
Ah sorry I hadn't refreshed the UI so I didn't see your follow up comment. Nvm!